opt/scopeserver/dataserver/utils/loom.py

```diff
@@ -31,9 +34,10 @@
 
 
 class Loom:
-    def __init__(self, partial_md5_hash, file_path, abs_file_path, loom_connection, loom_file_handler):
+    def __init__(
+        self, file_path: Path, abs_file_path: Path, loom_connection: LoomConnection, loom_file_handler,
```

Comment:
I'm not sure how to resolved the typing for loom_file_handler here. To use the `LoomFileHandler` class, we need to import the `loom_file_handler` module, which imports the `Loom` class from the `loom` module. This ends up being circular and everything breaks.

Reply:
They recommend refactoring the code to avoid these circular dependencies (probably the best). Otherwise you could use this trick: https://www.stefaanlippens.net/circular-imports-type-hints-python.html

Reply:
As discussed on Skype, this refactoring will come _after_ we decide on the new format for the loom files, as this is going to require a lot of work here also.